storage-bigtable/src/lib.rs

```diff
@@ -399,9 +404,34 @@ impl Default for LedgerStorageConfig {
     }
 }
 
+const METRICS_REPORT_INTERVAL_MS: u64 = 10_000;
+
+#[derive(Default)]
+struct LedgerStorageStats {
+    num_queries: AtomicUsize,
+    last_report: AtomicInterval,
+}
+
+impl LedgerStorageStats {
+    fn update_and_maybe_report(&self) {
```

Comment:
> I think it is kind of sneaky that a method called this actually increments a field called num_queries under the hood. Plus LedgerStorageStats is a pretty generic name, suggesting it might be used for accumulating and reporting other stats too in the future.

Fair points. As you correctly identified, I gave it a generic name incase we wanted to expand in the future. I don't have anything in mind at the moment tho

> Perhaps this could be called increment_num_queries() instead, and call a separate maybe_report() method inside it?

I'm on board with this. Hypothetically if we did add some other field(s), we would need separate accessors for updating those so seems reasonable to make the setter name specific to that field